File: attribouter/src/main/java/me/jfenn/attribouter/dialogs/LicenseDialog.java
package me.jfenn.attribouter.dialogs;

import android.content.Context;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatDialog;
import android.view.View;
import android.widget.TextView;

import me.jfenn.attribouter.R;
import me.jfenn.attribouter.wedges.LicenseWedge;
import me.jfenn.attribouter.utils.ResourceUtils;
import me.jfenn.attribouter.utils.UrlClickListener;

public class LicenseDialog extends AppCompatDialog {

    private LicenseWedge license;

    public LicenseDialog(Context context, LicenseWedge license) {
        super(context, ResourceUtils.getThemeResourceAttribute(context, R.styleable.AttribouterTheme_licenseDialogTheme, R.style.AttribouterTheme_Dialog));
        this.license = license;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dialog_attribouter_license);

        TextView nameView = findViewById(R.id.name);
        TextView descriptionView = findViewById(R.id.description);
        View infoContainerView = findViewById(R.id.infoContainer);
        View permissionsView = findViewById(R.id.permissions);
        TextView permissionsTextView = findViewById(R.id.permissionsText);
        View conditionsView = findViewById(R.id.conditions);
        TextView conditionsTextView = findViewById(R.id.conditionsText);
        View limitationsView = findViewById(R.id.limitations);
        TextView limitationsTextView = findViewById(R.id.limitationsText);
        View bodyContainerView = findViewById(R.id.bodyContainer);
        TextView bodyView = findViewById(R.id.body);
        View moreInfoButton = findViewById(R.id.moreInfo);

        nameView.setText(ResourceUtils.getString(getContext(), license.licenseName));
        if (license.licenseDescription != null)
            descriptionView.setText(license.licenseDescription);
        else descriptionView.setVisibility(View.GONE);

        String permissions = license.getLicensePermissions();
        if (permissions != null)
            permissionsTextView.setText(permissions);
        else permissionsView.setVisibility(View.GONE);

        String conditions = license.getLicenseConditions();
        if (conditions != null)
            conditionsTextView.setText(conditions);
        else conditionsView.setVisibility(View.GONE);

        String limitations = license.getLicenseLimitations();
        if (limitations != null)
            limitationsTextView.setText(limitations);
        else limitationsView.setVisibility(View.GONE);

        if (permissions == null && conditions == null && limitations == null)
            infoContainerView.setVisibility(View.GONE);

        String body = ResourceUtils.getString(getContext(), license.licenseBody);
        if (body != null)
            bodyView.setText(body);
        else bodyContainerView.setVisibility(View.GONE);

        String moreInfo = ResourceUtils.getString(getContext(), license.licenseUrl);
        if (moreInfo != null)
            moreInfoButton.setOnClickListener(new UrlClickListener(moreInfo));
        else moreInfoButton.setVisibility(View.GONE);
    }
}

File: attribouter/src/main/java/me/jfenn/attribouter/dialogs/LicenseDialog.kt
package me.jfenn.attribouter.dialogs

import android.content.Context
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatDialog
import androidx.appcompat.widget.Toolbar
import me.jfenn.androidutils.autoSystemUiColors
import me.jfenn.androidutils.bind
import me.jfenn.attribouter.R
import me.jfenn.attribouter.utils.ResourceUtils.getString
import me.jfenn.attribouter.utils.UrlClickListener
import me.jfenn.attribouter.utils.getThemeAttr
import me.jfenn.attribouter.utils.toListString
import me.jfenn.attribouter.wedges.LicenseWedge

class LicenseDialog(
        context: Context,
        private val license: LicenseWedge
) : AppCompatDialog(
        context,
        context.getThemeAttr(R.attr.attribouter_licenseDialogTheme, R.style.AttribouterTheme_Dialog_Fullscreen)
) {

    val toolbar: Toolbar? by bind(R.id.toolbar)
    val nameView: TextView? by bind(R.id.name)
    val descriptionView: TextView? by bind(R.id.description)
    val permissionsTextView: TextView? by bind(R.id.permissionsText)
    val conditionsTextView: TextView? by bind(R.id.conditionsText)
    val limitationsTextView: TextView? by bind(R.id.limitationsText)
    val bodyView: TextView? by bind(R.id.body)
    val moreInfoButton: View? by bind(R.id.moreInfo)

    override fun onStart() {
        super.onStart()
        window?.autoSystemUiColors()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.attribouter_dialog_license)

        val licenseName = getString(context, license.licenseName)

        toolbar?.apply {
            title = licenseName
            setNavigationOnClickListener { dismiss() }
        }

        nameView?.text = licenseName
        descriptionView?.text = getString(context, license.licenseDescription)

        license.licensePermissions?.let { permissions ->
            permissionsTextView?.text = permissions.toListString()
        }

        license.licenseConditions?.let { conditions ->
            conditionsTextView?.text = conditions.toListString()
        }

        license.licenseLimitations?.let { limitations ->
            limitationsTextView?.text = limitations.toListString()
        }

        bodyView?.text = getString(context, license.licenseBody)?.trimIndent()

        val moreInfo = getString(context, license.licenseUrl)
        moreInfo?.let {
            moreInfoButton?.setOnClickListener(UrlClickListener(it))
        } ?: run {
            moreInfoButton?.visibility = View.GONE
        }
    }

}